src/main/java/com/onaio/steps/helper/UploadFileTask.java

```diff
@@ -67,36 +68,53 @@ public void upload(File file) {
-
                 RequestBody surveyIdBody = RequestBody.create(MediaType.parse("text/plain"), surveyId);
                 RequestBody userIdBody = RequestBody.create(MediaType.parse("text/plain"), userId);
                 RequestBody userPasswordBody = RequestBody.create(MediaType.parse("text/plain"), userPassword);
 
-                retrofit.create(HouseholdService.class).uploadData(endpoint, fileBody, surveyIdBody, userIdBody, userPasswordBody).enqueue(new Callback<ResponseBody>() {
-                    @Override
-                    public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
-                        if (response.isSuccessful() && response.code() == 201) {
-                            new CustomNotification().notify(activity, R.string.export_complete, R.string.export_complete_message);
-                            onExportListener.onFileUploaded(true);
-                        } else {
-                            new CustomNotification().notify(activity, R.string.error_title, R.string.export_failed);
-                            onExportListener.onFileUploaded(false);
-                        }
-                    }
-
-                    @Override
-                    public void onFailure(Call<ResponseBody> call, Throwable t) {
-                        new CustomNotification().notify(activity, R.string.error_title, R.string.export_failed);
-                        onExportListener.onFileUploaded(false);
-                    }
-                });
+                if (!files.isEmpty()) {
+                    upload(files, endPoint, surveyIdBody, userIdBody, userPasswordBody);
+                }
             }
         } else {
             new CustomNotification().notify(activity, R.string.error_title, R.string.export_failed);
             onExportListener.onFileUploaded(false);
         }
     }
+
+    public void upload(Queue<File> files, String endPoint, RequestBody surveyIdBody, RequestBody userIdBody, RequestBody userPasswordBody) {
+
+        File file = files.remove();
+        String fileType = "text/csv";
+        RequestBody requestFile = RequestBody.create(MediaType.parse(fileType), file);
+        MultipartBody.Part fileBody = MultipartBody.Part.createFormData("file", file.getName(), requestFile);
+
+        retrofit.create(HouseholdService.class).uploadData(endPoint, fileBody, surveyIdBody, userIdBody, userPasswordBody).enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                if (response.isSuccessful() && response.code() == 201) {
+                    if (files.isEmpty()) {
+                        new CustomNotification().notify(activity, R.string.export_complete, R.string.export_complete_message);
+                        onExportListener.onFileUploaded(true);
```

Comment:
1. We need to be able to handle each upload as a unit and therefore marking only the households in that upload as sent. The above line means and Line 99-100 does not handle this. It only handles when all the files are uploaded
2. We have another notification call (`new CustomNotification().notify()`) inside the listener here https://github.com/onaio/steps-app/blob/7dddc9c073f03f6075f86767efe62df6599161ea/src/main/java/com/onaio/steps/handler/actions/SubmitDataHandler.java#L104

Reply:
I've updated the logic and now holding each response as a `UploadResult` and finally show a dialog with a list of success and failure forms.